### PR TITLE
Add direction to Swipeable callbacks

### PR DESCRIPTION
**Pull request by jakub-gonet**
## Description

Added direction to `onSwipeable(Will)Open` & `onSwipeable(Will)Close`.

Should we deprecate `onSwipeable(Left/Right)Open` and similar callbacks?

## Test plan

Added console.logs to `AppleStyleSwipeableRow`

### CODE REVIEW

File: example/src/showcase/swipeable/AppleStyleSwipeableRow.tsx [hunk 1]
```diff
@@ -96,7 +96,13 @@ export default class AppleStyleSwipeableRow extends Component {
         leftThreshold={30}
         rightThreshold={40}
         renderLeftActions={this.renderLeftActions}
-        renderRightActions={this.renderRightActions}>
+        renderRightActions={this.renderRightActions}
+        onSwipeableOpen={(direction) => {
+          console.log(`Opening swipeable to the ${direction}`);
```
Comment (j-piasecki) [inline example/src/showcase/swipeable/AppleStyleSwipeableRow.tsx:101]: A very minor nitpick, could you change this message in some way, for example:
```suggestion
          console.log(`Opening swipeable from the ${direction}`);
```
I was confused for a moment when I swiped from left to right and the console log said that it was opening to the left.

File: src/components/Swipeable.tsx [hunk 2]
```diff
@@ -102,12 +102,12 @@ export interface SwipeableProps
   /**
    * Called when action panel gets open (either right or left).
    */
-  onSwipeableOpen?: () => void;
+  onSwipeableOpen?: (direction: 'left' | 'right') => void;
```
Comment (j-piasecki) [inline src/components/Swipeable.tsx:105]: We already have a `Directions` enum, although it also provides `UP` and `DOWN` directions. Do you think using it here is fine or the fact that it has all four directions would be a problem?

**Review by j-piasecki — APPROVED** [review]
IMO we can deprecate those methods, they provide exactly the same functionality now.

File: src/components/Swipeable.tsx (hunk 2 above)
Comment (jakub-gonet) [reply src/components/Swipeable.tsx:105]: Swipeable doesn't open to vertical directions so typing it with Directions is wrong here. I'd like to stay with the current typing.

Side note: we can move Directions to FlingGH definition file - from what I saw it's used only there.

File: example/src/showcase/swipeable/AppleStyleSwipeableRow.tsx (hunk 1 above)
Comment (jakub-gonet) [reply example/src/showcase/swipeable/AppleStyleSwipeableRow.tsx:101]: Sure, will do.

Comment (jakub-gonet) [reply example/src/showcase/swipeable/AppleStyleSwipeableRow.tsx]: Fixed in 5438419.

**Pull request merged**